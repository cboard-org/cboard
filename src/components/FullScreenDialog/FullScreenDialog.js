import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { withStyles } from 'material-ui/styles';
import AppBar from 'material-ui/AppBar';
import Button from 'material-ui/Button';
import Dialog, { DialogContent } from 'material-ui/Dialog';
import Paper from 'material-ui/Paper';
import Toolbar from 'material-ui/Toolbar';
import Typography from 'material-ui/Typography';
import IconButton from 'material-ui/IconButton';
import ArrowBackIcon from 'material-ui-icons/ArrowBack';
import Slide from 'material-ui/transitions/Slide';

import messages from '../../containers/App/messages';

const styles = {
  appBar: {
    position: 'static',
    flexShrink: 0
  },
  title: {
    flex: 1
  },
  container: {
    height: '100%',
    background: '#f1f1f1',
    overflowY: 'auto',
    WebkitOverflowScrolling: 'touch'
  },
  paper: {
    background: '#fff',
    maxWidth: '680px',
    margin: '0 auto'
  }
};

function FullScreenDialog(props) {
  const { open, title, onCancel, onSubmit, classes, children } = props;
  return (
    <Dialog
      fullScreen
      open={open}
      transition={<Slide direction="up" />}
      onRequestClose={() => {
        onCancel();
      }}
    >
      <AppBar className={classes.appBar}>
        <Toolbar>
          <div className="back-button">
            <IconButton
              color="contrast"
              onClick={() => {
                onCancel();
              }}
            >
              <ArrowBackIcon />
            </IconButton>
          </div>
          <Typography type="title" color="inherit" className={classes.title}>
            {title}
          </Typography>
          {onSubmit &&
            <Button
              color="contrast"
              onClick={() => {
                onSubmit();
                onCancel();
              }}
            >
              <FormattedMessage {...messages.save} />
            </Button>}
        </Toolbar>
      </AppBar>
      <div className={classes.container}>
        <Paper className={classes.paper}>
          <DialogContent>
            {children}
          </DialogContent>
        </Paper>
      </div>
    </Dialog>
  );
}

FullScreenDialog.propTypes = {
  open: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  onCancel: PropTypes.func,
  onSubmit: PropTypes.func,
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  children: PropTypes.node
};

FullScreenDialog.defaultProps = {
  onCancel: () => {}
};

export default withStyles(styles, { name: 'FullScreenDialog' })(
  FullScreenDialog
);
