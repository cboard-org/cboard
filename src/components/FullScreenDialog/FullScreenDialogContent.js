import React from 'react';
import { DialogContent } from 'material-ui/Dialog';

function FullScreenDialogContent({ children }) {
  return <DialogContent>{children}</DialogContent>;
}

export { FullScreenDialogContent };
