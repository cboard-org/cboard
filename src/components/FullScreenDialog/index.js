export { default } from './FullScreenDialog';
export { FullScreenDialogContent } from './FullScreenDialogContent';
