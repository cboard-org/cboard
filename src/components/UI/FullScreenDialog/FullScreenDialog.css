.FullScreenDialog__title {
  flex: 0 1 auto;
  overflow: hidden;
}

[dir='ltr'] .FullScreenDialog__title {
  margin-right: auto;
}

[dir='rtl'] .FullScreenDialog__title {
  margin-left: auto;
}

.FullScreenDialog__buttons {
  height: 48px;
  display: flex;
  flex: 1;
  justify-content: flex-end;
}

.FullScreenDialogContent {
  padding: 24px 16px;
}
