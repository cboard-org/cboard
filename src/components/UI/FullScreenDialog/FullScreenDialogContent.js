import React from 'react';
import classNames from 'classnames';

function FullScreenDialogContent({ className, children }) {
  const fullScreenDialogContentClassName = classNames(
    'FullScreenDialogContent',
    className
  );
  return <div className={fullScreenDialogContentClassName}>{children}</div>;
}

export { FullScreenDialogContent };
