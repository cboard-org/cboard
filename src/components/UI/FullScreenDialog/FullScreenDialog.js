import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { withStyles, useTheme } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Slide from '@material-ui/core/Slide';
import Fade from '@material-ui/core/Fade';

import messages from '../../App/App.messages';
import BackButton from '../BackButton';
import './FullScreenDialog.css';

FullScreenDialog.propTypes = {
  disableSubmit: PropTypes.bool,
  open: PropTypes.bool,
  fullWidth: PropTypes.bool,
  title: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  onClose: PropTypes.func,
  onSubmit: PropTypes.func,
  classes: PropTypes.object.isRequired,
  className: PropTypes.string,
  children: PropTypes.node
};

FullScreenDialog.defaultProps = {
  onClose: () => {}
};

const styles = {
  appBar: {
    position: 'static',
    flexShrink: 0
  },
  title: {
    flex: 1,
    whiteSpace: 'nowrap',
    textOverflow: 'ellipsis',
    overflow: 'hidden'
  },
  container: {
    background: '#f1f1f1',
    height: '100%',
    overflowY: 'auto',
    WebkitOverflowScrolling: 'touch'
  },
  containerDark: {
    background: '#1111',
    height: '100%',
    overflowY: 'auto',
    WebkitOverflowScrolling: 'touch'
  },
  content: {
    maxWidth: '680px',
    margin: '0 auto'
  },
  contentFullWidth: {
    margin: '0 auto'
  }
};

const transitions = {
  UP: 'up',
  FADE: 'fade'
};

const TransitionUp = React.forwardRef((props, ref) => (
  <Slide direction="up" {...props} ref={ref} />
));
const TransitionFade = React.forwardRef((props, ref) => (
  <Fade {...props} ref={ref} />
));

function getTransition(transition) {
  switch (transition) {
    case transitions.UP:
      return TransitionUp;
    case transitions.FADE:
      return TransitionFade;
    default:
    // no default
  }
}

function FullScreenDialog(props) {
  const {
    classes,
    children,
    open,
    title,
    buttons,
    disableSubmit,
    onClose,
    onSubmit,
    transition = transitions.UP,
    fullWidth
  } = props;

  const theme = useTheme();
  const dark = theme.palette.type === 'dark' ? true : false;

  return (
    <Dialog
      fullScreen
      open={open}
      TransitionComponent={getTransition(transition)}
      onClose={onClose}
    >
      <AppBar className={classes.appBar}>
        <Toolbar disableGutters>
          <BackButton onClick={onClose} />

          {title && (
            <div className="FullScreenDialog__title">
              <Typography
                variant="h6"
                color="inherit"
                className={classes.title}
              >
                {title}
              </Typography>
            </div>
          )}
          {buttons && (
            <div className="FullScreenDialog__buttons">{buttons}</div>
          )}
          {onSubmit && (
            <Button
              id="save-button"
              disabled={disableSubmit}
              color="inherit"
              style={{ color: disableSubmit && 'rgba(255, 255, 255, 0.26)' }}
              onClick={() => {
                onSubmit();
                onClose();
              }}
            >
              <FormattedMessage {...messages.save} />
            </Button>
          )}
        </Toolbar>
      </AppBar>
      <div className={dark ? classes.containerDark : classes.container}>
        <div className={fullWidth ? classes.contentFullWidth : classes.content}>
          {children}
        </div>
      </div>
    </Dialog>
  );
}

export default withStyles(styles, { name: 'FullScreenDialog' })(
  FullScreenDialog
);
