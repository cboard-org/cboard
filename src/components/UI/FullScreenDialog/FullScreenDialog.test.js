import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import FullScreenDialog from './FullScreenDialog';

jest.mock('../../App/App.messages', () => {
  return {
    save: {
      id: 'cboard.components.App.save',
      defaultMessage: 'Save'
    },
    newContentAvailable: {
      id: 'cboard.components.App.newContentAvailable',
      defaultMessage: 'New content is available; please refresh.'
    },
    contentIsCached: {
      id: 'cboard.components.App.contentIsCached',
      defaultMessage: 'Content is cached for offline use.'
    }
  };
});

const props = {
  disableSubmit: false,
  open: true,
  title: 'test',
  onClose: jest.fn(),
  onSubmit: jest.fn(),
  disableSubmit: false,
  transition: 'up'
};

describe('FullScreenDialog tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<FullScreenDialog {...props} />);
  });
  test('check transition', () => {
    const fade = { ...props, transition: 'fade' };
    const wrapper = shallow(<FullScreenDialog {...fade} />);
  });
});
