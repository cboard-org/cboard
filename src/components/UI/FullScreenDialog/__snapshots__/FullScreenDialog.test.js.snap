// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenDialog tests default renderer 1`] = `
<FullScreenDialog
  classes={
    Object {
      "appBar": "FullScreenDialog-appBar-1",
      "container": "FullScreenDialog-container-3",
      "content": "FullScreenDialog-content-4",
      "title": "FullScreenDialog-title-2",
    }
  }
  disableSubmit={false}
  onClose={[MockFunction]}
  onSubmit={[MockFunction]}
  open={true}
  title="test"
  transition="up"
/>
`;
